tests/e2e_logging.py

```diff
@@ -245,7 +251,12 @@ def test_protocols(network, args):
         else:
             assert res.returncode != 0, res.returncode
             err = res.stderr.decode()
-            assert expected_error in err, err
+            failed = False
+            for err_i in expected_errors:
+                if err_i in err:
+                    failed = True
+                    break
+            assert failed, "none of the expected error messages were found"
```

Comment:
This doesn't print the original `err` message if none of the expected errors are found, which will make it harder to debug. It's possible to find `expected_errors` from the callstack + inspecting this source, but not the runtime value of `err`. We should log that on failure, and can do so with a list comprehension assertion:

```suggestion
            assert any(expected_error in err for expected_error in expected_errors), err
```

Reply:
Yeah, that's much better, thanks! 

@achamayou do you mean a set intersection non-empty check? Would that be shorter?